Source/inv.cpp

```diff
@@ -1109,6 +1110,37 @@ void CleanupItems(Item *item, int ii)
 	}
 }
 
+std::optional<Point> FindClosestValidItemPosition(Point startingPosition)
+{
+	if (CanPut(startingPosition)) {
+		return startingPosition;
+	}
+	if (dItem[startingPosition.x][startingPosition.y] == 0)
+		return {};

```

Comment:
Another thing I noticed while testing... This check stops the player dropping items inside houses (which is great :D), but also stops ctrl+click dropping items if you're standing on a tile above an object (something where `IsItemBlockingObjectAtPosition` returns true). Relatively minor issue since the player can just move to a different tile, break/open the object, or drop with the mouse targeting a valid tile.